src/System.Diagnostics.Process/src/System/Diagnostics/Process.Linux.cs

```diff
@@ -0,0 +1,696 @@
+            // at the max size of IntPtr.  This often happens when there is no configured 
+            // rsslim other than ulong.MaxValue, which without these checks would show up
+            // as a maxWorkingSet == -1.
+            switch (IntPtr.Size)
+            {
+                case 4:
+                    if (rsslim > int.MaxValue)
+                        rsslim = int.MaxValue;
+                    break;
+                case 8:
+                    if (rsslim > long.MaxValue)
+                        rsslim = long.MaxValue;
+                    break;
+            }
+
+            maxWorkingSet = (IntPtr)rsslim;
+        }
+
+        /// <summary>Sets one or both of the minimum and maximum working set limits.</summary>
+        /// <param name="newMin">The new minimum working set limit, or null not to change it.</param>
+        /// <param name="newMax">The new maximum working set limit, or null not to change it.</param>
+        /// <param name="resultingMin">The resulting minimum working set limit after any changes applied.</param>
+        /// <param name="resultingMax">The resulting maximum working set limit after any changes applied.</param>
+        private void SetWorkingSetLimitsCore(IntPtr? newMin, IntPtr? newMax, out IntPtr resultingMin, out IntPtr resultingMax)
+        {
+            // RLIMIT_RSS with setrlimit not supported on Linux > 2.4.30.
+            throw new PlatformNotSupportedException();
+        }
+
+        /// <summary>Starts the process using the supplied start info.</summary>
+        /// <param name="startInfo">The start info with which to start the process.</param>
+        private bool StartCore(ProcessStartInfo startInfo)
+        {
+            // Resolve the path to the specified file name
+            string executablePath = ResolvePath(startInfo.FileName);
+
+            // TODO: Fix stream redirection.
+            // If streams are to be redirected, create a pipe for each that is.
+            // int[0] is the read end of the pipe, int[1] is the write end.
+            int[] stdinPipeFds = startInfo.RedirectStandardInput ? CreatePipe() : null;
+            int[] stdoutPipeFds = startInfo.RedirectStandardOutput ? CreatePipe() : null;
+            int[] stderrPipeFds = startInfo.RedirectStandardError ? CreatePipe() : null;
+
+            // Fork the process.  In the child process, pid will be 0.
+            // In the parent process, pid will be the child's process ID.
+            int pid = Interop.libc.fork();
+            if (pid == -1) // fork error
+            {
+                throw new Win32Exception();
+            }
+
+            if (pid == 0) // child process
+            {
+                // If we're redirecting streams, configure stdin, stdout, stderr
+                if (stdinPipeFds != null)
+                {
+                    Redirect(stdinPipeFds[0], Interop.libc.FileDescriptors.STDIN_FILENO);
```

Comment:
Consider having consts in this method for ReadEndOfPipe and WriteEndOfPipe to make the indexing more obvious?